src/Sequence.cpp

```diff
@@ -308,4 +324,49 @@ Sequence::createCommandBuffer()
     KP_LOG_DEBUG("Kompute Sequence Command Buffer Created");
 }
 
+void
+Sequence::createTimestampQueryPool(uint32_t query_size)
+{
+    KP_LOG_DEBUG("Kompute Sequence creating query pool");
+    if (!this->mDevice) {
+        throw std::runtime_error("Kompute Sequence device is null");
+    }
+    if (!this->mPhysicalDevice) {
+        throw std::runtime_error("Kompute Sequence physical device is null");
+    }
+
+    vk::PhysicalDeviceProperties physicalDeviceProperties =
+      this->mPhysicalDevice->getProperties();
+    
+    if(physicalDeviceProperties.limits.timestampComputeAndGraphics){
+      vk::QueryPoolCreateInfo queryPoolInfo;
+      queryPoolInfo.setQueryCount(query_size);
+      queryPoolInfo.setQueryType(vk::QueryType::eTimestamp);
+      this->timestampQueryPool = std::make_shared<vk::QueryPool>(this->mDevice->createQueryPool(queryPoolInfo));
+
+      KP_LOG_DEBUG("Query pool for timestamps created");
+    }
+    else{
+      KP_LOG_DEBUG("Device does not support timestamps");
```

Comment:
Update to warn or error given that it was requested and not possible to create

Reply:
Added exception